godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/Godot.kt

```diff
@@ -0,0 +1,87 @@
+package godot.core
+
+import godot.gdnative.*
+import kotlinx.cinterop.*
+import kotlin.native.concurrent.AtomicInt
+import kotlin.native.concurrent.AtomicReference
+
+internal object Godot {
+    private val gdnativeWrapper = AtomicReference<CPointer<godot_gdnative_core_api_struct>?>(null)
+    private val nativescriptWrapper = AtomicReference<CPointer<godot_gdnative_ext_nativescript_api_struct>?>(null)
+
+    internal val gdnative: godot_gdnative_core_api_struct
+        get() = checkNotNull(gdnativeWrapper.value).pointed
+
+    internal val gdnative11: godot_gdnative_core_1_1_api_struct
+        get() = checkNotNull(gdnative.next).reinterpret<godot_gdnative_core_1_1_api_struct>().pointed
+
+    internal val gdnative12: godot_gdnative_core_1_2_api_struct
+        get() = checkNotNull(gdnative11.next).reinterpret<godot_gdnative_core_1_2_api_struct>().pointed
+
+    @PublishedApi
+    internal val nativescript: godot_gdnative_ext_nativescript_api_struct
+        get() = checkNotNull(nativescriptWrapper.value).pointed
+
+    @PublishedApi
+    internal val nativescript11: godot_gdnative_ext_nativescript_1_1_api_struct
+        get() = checkNotNull(nativescript.next).reinterpret<godot_gdnative_ext_nativescript_1_1_api_struct>().pointed
+
+    internal val languageIndex: Int
+        get() = languageIndexRef.value
+
+    private val languageIndexRef = AtomicInt(-1)
+
+    private val initHandle = AtomicInt(0)
+
+    fun init(options: godot_gdnative_init_options) {
+        val gdnative = checkNotNull(options.api_struct)
+        val extensionCount = gdnative.pointed.num_extensions.toInt()
+        val extensions = checkNotNull(gdnative.pointed.extensions)
+        lateinit var nativescript: CPointer<godot_gdnative_ext_nativescript_api_struct>
+        (0 until extensionCount).forEach { i ->
+            val extension = checkNotNull(extensions[i])
+            val type = extension.pointed.type
+            when (GDNATIVE_API_TYPES.byValue(type)) {
+                GDNATIVE_API_TYPES.GDNATIVE_EXT_NATIVESCRIPT -> {
+                    nativescript = extension.reinterpret()
+                }
+                else -> {
```

Comment:
Nah, this case is totally okay since we only use a single gdnative extension. Otherwise, the log will always show when godot loads which doesn't affect the user and they can't do anything either.

Reply:
:+1: